### PR TITLE
Migrate Query Workbench to new Platform 

### PR DESCRIPTION
**Pull request by davidcui-amzn**
*Issue #, if available:*
N/A
*Description of changes:*
* Migrated query workbench to new Kibana platform. 
     * Refactored server routes and removed `sql_console.css` in place of `scss`
     * Removed `<Header />` from `Main`
     * Changed response type of backend services to return custom vs `ResponseToolkit` object 
     * Changed input type of backend services to intake single `queryString` parameter 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: workbench/public/components/Main/main.tsx [hunk 1]
```diff
@@ -64,7 +64,7 @@ export type ItemIdToExpandedRowMap = {
 };
 
 interface MainProps {
-  httpClient: IHttpService;
+  httpClient: any;
```
[the remark this answers is not in the document]
Comment (davidcui-amzn) [reply workbench/public/components/Main/main.tsx:67]: Good catch, forgot to update this- will change

File: workbench/public/components/Main/main.tsx [hunk 2]
```diff
@@ -181,8 +181,7 @@ export function getQueryResultsForTable(queryResults: ResponseDetail<string>[]):
 }
 
 export class Main extends React.Component<MainProps, MainState> {
-  httpClient: IHttpService;
-  // httpClient: any;
+  httpClient: any;
```
[the remark this answers is not in the document]
Comment (davidcui-amzn) [reply workbench/public/components/Main/main.tsx:184]: Same as above

File: workbench/public/components/app.tsx [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+export const WorkbenchApp = ({ basename, notifications, http, navigation }: WorkbenchAppDeps) => {
+  return (
+    <Router basename={'/' + basename}>
+      <I18nProvider>
+        <div>
+          <EuiPage>
+            <EuiPageBody>
+              <Route
+                path="/"
+                render={(props) => <Main httpClient={http} {...props} onChange={onChange} />}
```
[the remark this answers is not in the document]
Comment (davidcui-amzn) [reply workbench/public/components/app.tsx:45]: `onChange` is required props for `Main` but is also defined in `Main` so I just created the empty `onChange` to pass into `Main`

File: workbench/server/services/utils/constants.ts [hunk 4]
```diff
@@ -32,3 +34,12 @@ export const CLUSTER = {
   SQL: 'opendistro_sql',
   DATA: 'data',
 };
+
+export const convertQueryToString = (query: string | ParsedUrlQuery) => {
+  let index;
+  var queryString = "";
```
[the remark this answers is not in the document]
Comment (davidcui-amzn) [reply workbench/server/services/utils/constants.ts:40]: The request object has changed, and the query is not in an `Object` with each key containing a single character from the query e.g `{ {'0': 'S'}, {'1': 'E'}, {'2': 'L'}, ...}` so `convertQueryToString` converts it to a string containing the query

File: workbench/common/index.ts [hunk 5]
```diff
@@ -0,0 +1,2 @@
+export const PLUGIN_ID = 'workbench';
```
[the remark this answers is not in the document]
Comment (davidcui-amzn) [reply workbench/common/index.ts:1]: Good catch, will change and add header

File: workbench/public/components/app.tsx (hunk 3 above)
[the remark this answers is not in the document]
Comment (joshuali925) [reply workbench/public/components/app.tsx:45]: Main uses `this.onChange` which is defined in Main itself, you can remove the `onChange` in `MainProps` (line 68 in `Main.tsx`) so you don't have to pass in this empty function

**Review by joshuali925 — APPROVED** [review]

File: workbench/server/services/utils/constants.ts (hunk 4 above)
[the remark this answers is not in the document]
Comment (joshuali925) [reply workbench/server/services/utils/constants.ts:40]: That's a weird problem.. is the use of `var` necessary? can we use `let`?

[the remark this answers is not in the document]
Comment (davidcui-amzn) [reply workbench/server/services/utils/constants.ts:40]: Yeah the change to the `KibanaRequest` type made the object change completely. And yes good catch, will change to `let`

**Review by chloe-zh — APPROVED** [review]

**Pull request merged**